File: Problem123.java
public class Problem123 {
    int profit;
    public int maxProfit(int[] prices) {
        profit = maxTransaction(prices, 0, prices.length-1);
        int index1 = 0;
        int index2 = 2;
        //index2 must be >2 away from index1
        if (prices.length > 4){
            for (int i = index1; i <= prices.length - 5; i++){
                //for (int j = i + 2; )
                
                
                
            }
            
            
            
            
            
        }
        return profit;
    }
    public int maxTransaction(int[] prices, int index1, int index2){
        int m = 0;
        for (int i = index1; i < index2; i++){
            for (int j = i; j < index2 + 1; j++){
                m = Math.max(m, prices[j] - prices[i]);
            }
        }
        return m;
    }
}

File: Problem123.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using std::vector;
using namespace std;
class Problem123{
    int maxProfit (vector<int>& prices){
        vector<int> answer;
        int highest = INT_MAX, int lowest = INT_MIN;
        int profit1 = 0, int profit2 = 0;

        for (int i = 0; i < prices.size(); ++i){
            lowest = min(lowest, prices[i]);
            profit1 = max(profit1, prices[i] - lowest);
            answer.push_back(profit1);
        }
        for (int j = prices.size()-1; j >= 0; j--){
            highest = max(highest, prices[j]);
            profit2 = max(profit2, highest - prices[j]);
            answer[j] += profit2;
        }
        return *max_element(answer.begin(), answer.end());
    }
};